adal/src/main/java/com/microsoft/aad/adal/Oauth2.java

```diff
@@ -391,9 +404,17 @@ public AuthenticationResult getToken(String authorizationUrl)
 
                 // Check if we have code
                 if (result != null && result.getCode() != null && !result.getCode().isEmpty()) {
+                    if (!StringExtensions.isNullOrBlank(result.getAuthority())) {
+                        setTokenEndpoint(result.getAuthority() + DEFAULT_TOKEN_ENDPOINT);
```

Comment:
Look at this iteration, if you still think this needs to change, lets talk about the best way to do it